### PR TITLE
fix(engine): limt clause conjunction with where clause or group by clause

### LINKED ISSUE #2433: where xxx = xxx limit 1返回多条数据
**Bug Description**

image

**Expected Behavior**

**Steps to Reproduce**

1. 
2. 
3. 
4.

### PR DESCRIPTION
**Pull request by aceforeverd**
fixes:
- limit of where clause
- limit of group by (+ having) clause
- correctness of `limit 0`: should returns empty list

close #2433 

the pr introduce a new iterator (`LimitIterator`) used for limit clause conjunction with where clause. The iterator is not seekable, so window project over this is not supported as well, we need the compile time check for this.

some restrictions should noted:
- limit clause do not support in request mode, I will add compile check later
- by practice, limit clause should used together with order by, while we do not support order by yet. As a result, limit result may not deterministic

### CODE REVIEW

File: hybridse/src/vm/catalog_wrapper.h [hunk 1]
```diff
@@ -337,56 +377,92 @@ class TableProjectWrapper : public TableHandler {
...
+    }
+
+    const Types& GetTypes() override { return table_hander_->GetTypes(); }
+    const IndexHint& GetIndex() override { return table_hander_->GetIndex(); }
+    const Schema* GetSchema() override { return table_hander_->GetSchema(); }
+    const std::string& GetName() override { return table_hander_->GetName(); }
+    const std::string& GetDatabase() override { return table_hander_->GetDatabase(); }
+
+    // FIXME(ace): do not use this, not implemented
+    std::shared_ptr<PartitionHandler> GetPartition(const std::string& index_name) override {
```
Comment (aceforeverd) [inline hybridse/src/vm/catalog_wrapper.h:461]: add the warning log

File: hybridse/src/vm/catalog_wrapper.h [hunk 2]
```diff
@@ -89,6 +90,44 @@ class IteratorFilterWrapper : public RowIterator {
...
+ public:
+    explicit LimitIterator(std::unique_ptr<RowIterator>&& iter, int32_t limit)
+        : RowIterator(), iter_(std::move(iter)), limit_(limit) {}
+    virtual ~LimitIterator() {}
+
+    bool Valid() const override {
+        return iter_->Valid() && cnt_ <= limit_;
+    }
+    void Next() override {
+        iter_->Next();
```
Comment (aceforeverd) [inline hybridse/src/vm/catalog_wrapper.h:107]: recheck this

File: cases/query/limit.yaml [hunk 3]
```diff
@@ -0,0 +1,352 @@
+# SQL limit clause
+# Syntax:
+#   `LIMIT <NUMBER>`
+#
+# Supported conjunction with
+# - where clause
+# - window projec
+# - last join
```
Comment (aceforeverd) [inline cases/query/limit.yaml:9]: more docs

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Comment by aceforeverd** [comment]
the pr also correct behavior of `limit 0`, @tobegit3hub pls recheck it :)

**Review by zhanghaohit — APPROVED** [review]
LGTM

**Review by tobegit3hub — APPROVED** [review]
LGTM

**Pull request merged**